assets/javascripts/application.js

```diff
@@ -0,0 +1,55 @@
+// Define error messages
+const error_message_length =
+  `<strong>Too Long</strong>: This form accepts a maximum of 1500 characters of text.
+  Please reduce the amount of text and try again.`
```

Comment:
What's the reason for the 1500 character limit? It's a practical limit for a demo, I suppose, but maybe someone wants to make sure rouge can parse 1,000,000 lines of text, let alone 1m characters. I also don't think the server/bandwidth would choke, but maybe I'm wrong! 😄

Reply:
Oh wow—thanks for taking the time to review this!

The reason I thought maybe we should have some kind of length limit (it's really a limit on the length of the request being greater than 2,000 but since that includes all the overhead, I just rounded it down quite a bit) was to prevent one or more users from over-consumption of the resources available in Heroku's free tier. I'm not sure how realistic a concern that is, though. Do you have much experience in that area? We have one worker and 1,000 dyno hours.

Reply:
Ah, I didn't know about a request length. The 1,000 hours shouldn't be a problem if this is the only app on Jeanine's account. It's 1,000/month and even on the longest months, that's only 744 hours (31 days * 24 hours).

Reply:
If somebody wants to test a million lines of text, they can use the CLI tool on their own machine IMO. This is for tinkering and bug repros.

Speaking of bug repros, the old site had a feature to use different versions of Rouge - have you copied that over?

Reply:
@jneen I haven't, no. I figured that since the app doesn't allow a user to save or share a snippet in any way, there wasn't really any utility to it. Would you still use it?

Reply:
I think it's definitely useful, and the sharing of snippets was one of the more useful bits - it allowed us to have consistent reproducible bug reports!

Have you considered encoding the snippet into the GET parameters? Then we could enable sharing with no database.

Reply:
I remembered we'd talked about it but when I was doing this initial version the additional complexity didn't seem worth it (read: the way it works in the current site's codebase looked cool but was a bit too clever for me to take in quickly). Snippets on the current site have been broken for about six months now and people just attach screenshots to their bug reports and it works well enough.

I think what I might do is might merge this change in and then take a crack at implementing it to see how hard it is.